openedx/frontend-lib-content-components
# Functional simple CAPA problem editor

This PR helps to get the CAPA problem editor working, this PR has:

1. Setting Widget
2. Question Widget
3. Answer Widget
4. OLX to React state Parser
5. React State to OLX Parser

This has the ability to support:

1. Numeric Problem
2. Text Input Problem
3. Dropdown Problem
4. Single Choice Problem
5. Multiple Choice Problem


~~**Discussions**: Link to any public dicussions about this PR or the design/architecture. Otherwise omit this.~~

**Dependencies**: None

**Screenshots**: 

![image](https://user-images.githubusercontent.com/7670449/201167366-edd2fc95-c2d5-4cbd-9d7b-41398acd7e5f.png)


**Sandbox URL**: TBD - sandbox is being provisioned.

**Merge deadline**: "None" if there's no rush, "ASAP" if it's critical, or provide a specific date if there is one.

**Testing instructions**:

1. Get the master branch working with LMS, Studio, and Course Authoring MFE and clone this repo in `{DEVSTACK}/src`
7. Change the branch to `farhaan/problem-editor-work`.
8. There are two ways to view the work `Gallery View` and `Studio`
9. For studio to work, add `new_core_editors.use_new_problem_editor` on http://localhost:18000/admin/waffle/flag/10/change/
10. And activate the waffle flag.
11. Now on studio add `Problem` and edit the problem, it should redirect to Course Authoring MFE.
12. The other way is to use Gallery View, get in the `frontend-lib-content-components/www`, run `npm install`, `npm start`.
13. Visit http://localhost:8080/index.html and you should see the problem rendered.

**Author notes and concerns**:
~~The settings for Problem Xblock API are not mapped yet.~~
~~The component are still in crude shape but it's getting there.~~
~~There are debugger statements and Extra information rendered on the webpage.~~

1. We still need to figure out if we need to have widgets to support `label` and `description` in problem OLX
2. The tolerance for Numeric input doesn't have a way to set in the UI.
3. The type in text input problem `ci` and `regex-ci` doesn't have a way to set up.

Additional improvements is:

1. Finding a way to have multiple numeric and text input boxes for the problem.


**Reviewers**
- [ ] @kaustavb12

Fixes: [BB-6631](https://tasks.opencraft.com/browse/BB-6631)
Co-authored-by: @kaustavb12 @navinkarkera 
Private-ref: [BB-6631](https://tasks.opencraft.com/browse/BB-6631)
